File: go-client-sdk/pkg/models/operations/patchcompaniescompanyidconnectionsconnectionidconnectioninfobankfeedaccountsbankaccountid.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
)

type PatchCompaniesCompanyIDConnectionsConnectionIDConnectionInfoBankFeedAccountsBankAccountIDPathParams struct {
	BankAccountID string `pathParam:"style=simple,explode=false,name=bankAccountId"`
	CompanyID     string `pathParam:"style=simple,explode=false,name=companyId"`
	ConnectionID  string `pathParam:"style=simple,explode=false,name=connectionId"`
}

type PatchCompaniesCompanyIDConnectionsConnectionIDConnectionInfoBankFeedAccountsBankAccountIDSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type PatchCompaniesCompanyIDConnectionsConnectionIDConnectionInfoBankFeedAccountsBankAccountIDRequest struct {
	PathParams PatchCompaniesCompanyIDConnectionsConnectionIDConnectionInfoBankFeedAccountsBankAccountIDPathParams
	Request    *shared.CodatStandardizationBankFeedsAccountsContractBankFeedBankAccount `request:"mediaType=application/json"`
	Security   PatchCompaniesCompanyIDConnectionsConnectionIDConnectionInfoBankFeedAccountsBankAccountIDSecurity
}

type PatchCompaniesCompanyIDConnectionsConnectionIDConnectionInfoBankFeedAccountsBankAccountIDResponse struct {
	CodatStandardizationBankFeedsAccountsContractBankFeedBankAccount *shared.CodatStandardizationBankFeedsAccountsContractBankFeedBankAccount
	ContentType                                                      string
	StatusCode                                                       int
}

File: typescript-client-sdk/src/sdk/models/operations/patchcompaniescompanyidconnectionsconnectionidconnectioninfobankfeedaccountsbankaccountid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class PatchCompaniesCompanyIdConnectionsConnectionIdConnectionInfoBankFeedAccountsBankAccountIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=bankAccountId" })
  bankAccountId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=companyId" })
  companyId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=connectionId" })
  connectionId: string;
}


export class PatchCompaniesCompanyIdConnectionsConnectionIdConnectionInfoBankFeedAccountsBankAccountIdSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeApiKeyAuth;
}


export class PatchCompaniesCompanyIdConnectionsConnectionIdConnectionInfoBankFeedAccountsBankAccountIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: PatchCompaniesCompanyIdConnectionsConnectionIdConnectionInfoBankFeedAccountsBankAccountIdPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: shared.CodatStandardizationBankFeedsAccountsContractBankFeedBankAccount;

  @SpeakeasyMetadata()
  security: PatchCompaniesCompanyIdConnectionsConnectionIdConnectionInfoBankFeedAccountsBankAccountIdSecurity;
}


export class PatchCompaniesCompanyIdConnectionsConnectionIdConnectionInfoBankFeedAccountsBankAccountIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatStandardizationBankFeedsAccountsContractBankFeedBankAccount?: shared.CodatStandardizationBankFeedsAccountsContractBankFeedBankAccount;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
